aten/src/ATen/ThreadLocalState.cpp

```diff
@@ -8,32 +8,28 @@
 
 namespace at {
 
-ThreadLocalState::ThreadLocalState(bool keep_grad_mode)
+ThreadLocalState::ThreadLocalState()
     : dispatch_key_(c10::impl::tls_local_dispatch_key_set()),
       debug_info_(c10::ThreadLocalDebugInfo::current()),
       autograd_tls_(c10::AutogradTLS::get_mode()) {
   rf_tls_ = at::get_record_function_tls_();
-
-#if !defined(CAFFE2_IS_XPLAT_BUILD) && !defined(C10_MOBILE)
```

Comment:
I guess this PR won't affect mobile/xplat builds, but just curious why we had this exception previously.

Reply:
My guess is that the mobile build doesn't come with cuda so trying to set the GradMode would raise an error there.